docs/js/features/odata/generate-client.md

```diff
@@ -0,0 +1,89 @@
+---
+id: generator-js-sdk
+title: Generate an OData client for JavaScript
+hide_title: false
+hide_table_of_contents: false
+sidebar_label: Generate an OData client for JavaScript
+keywords:
+- sap
+- cloud
+- sdk
+- odata
+- JavaScript 
+- TypeScript
+- generate
+author: Charles Dubois
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Generate a typed OData client with the SAP Cloud SDK generator
+
+The generator allows you to generate custom OData clients libraries for OData services. You can then easily access these services via the client libraries.
+
+The SAP Cloud SDK generator can both be used as a command line interface (CLI) and programmatically.
+
+All you need to use it is a service metadata specification in the `EDMX` format (file ending can be `.edmx` or `.xml`).
+
+## Installation
+
+Run this command in your project's terminal to install the generator globally.
+```sh
+npm install -g @sap-cloud-sdk/generator
+```
+
+:::note
+The generator is included in the SAP Cloud SDK CLI, if you have it installed you don't need to install the generator separately.
+:::
+
+## Usage
+
+### From the Command Line
+
+The generator is primarily meant to be used on the command line:
```

Comment:
If you still find some time consider giving this a bit more life ;) Maybe in the direction of:

The SAP Cloud SDK generator is primarily intended to be used on the command line. Run

```sh
generate-odata-client --inputDir path/to/your/service-specifications --outputDir path/to/store/generated/modules
```

This will generate OData clients for all your service specifications within the `path/to/your/service-specifications` directory in `EDMX` format and write the to the ` path/to/store/generated/modules` directory.


But this is just from the top of my head. I think at least explaining that those are the mandatory parameters and what they do in textual form might be helpful for understanding what to do.